src/kernel/syscalls.zig

```diff
@@ -0,0 +1,62 @@
+const std = @import("std");
+const scheduler = @import("scheduler.zig");
+const panic = @import("panic.zig").panic;
+const log = std.log.scoped(.syscalls);
+
+var allocator: *std.mem.Allocator = undefined;
+
+pub const Error = error{
+    OutOfMemory,
+};
+
+pub fn fromErrorCode(code: usize) Error {
+    return switch (code) {
+        1 => Error.OutOfMemory,
+        else => unreachable,
+    };
+}
+
+pub fn toErrorCode(err: Error) usize {
+    return switch (err) {
+        Error.OutOfMemory => 1,
+    };
+}
+
+comptime {
+    inline for (@typeInfo(Error).ErrorSet.?) |err| {
+        const error_instance = @field(Error, err.name);
+        if (fromErrorCode(toErrorCode(error_instance)) != error_instance) {
+            @compileError("toErrorCode and fromErrorCode are not synchronised for syscall error '" ++ err.name ++ "'\n");
+        }
+        inline for (@typeInfo(Error).ErrorSet.?) |err2| {
+            const error2_instance = @field(Error, err2.name);
+            if (error_instance != error2_instance and toErrorCode(error_instance) == toErrorCode(error2_instance)) {
+                @compileError("Syscall errors '" ++ err.name ++ "' and '" ++ err2.name ++ "' share the same error code\n");
+            }
+        }
+    }
+}
+
+pub const Syscall = enum {
+    Void,
```

Comment:
This is just here as an example so the compiler doesn't complain that the enum is empty. I'll be replacing it with proper syscalls once this PR is merged.

Reply:
Done